### PR TITLE
add dependency review action

### PR DESCRIPTION
**Pull request by cfkoehler**
Got this as a recommended action for PR's and since I am going through and updating actions might as well add another one!

### CODE REVIEW

File: .github/workflows/dependency-review.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  contents: read
+
+jobs:
+  dependency-review:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Checkout Repository'
+        uses: actions/checkout@v3
+      - name: 'Dependency Review'
+        uses: actions/dependency-review-action@v2
```
Comment (dev-mlb) [inline .github/workflows/dependency-review.yml:20]: Looks like there is a v3, is there a reason to stick with v2?

Comment (cfkoehler) [inline .github/workflows/dependency-review.yml:20]:
```suggestion
        uses: actions/dependency-review-action@v3
```

**Review by dev-mlb — APPROVED** [review]

**Comment by ldhardy** [comment]
I think it will be necessary to modify default configs for this action.

The severity defaults to low, I'm not sure we're going to want that?  Or maybe we do?
```
with:
          fail-on-severity: moderate
```

Is there  a way to see an example of this before it gets merged?  Can I branch your personal branch and create a PR back to your branch to see it run?

**Comment by cfkoehler** [comment]
> Is there a way to see an example of this before it gets merged? Can I branch your personal branch and create a PR back to your branch to see it run?

This PR has the action run showing no dependency changes: https://github.com/NationalSecurityAgency/emissary/actions/runs/4638563872

PR showing new vulnerable dependencies: https://github.com/NationalSecurityAgency/emissary/actions/runs/4775533594/jobs/8489903135?pr=467
Screenshot 2023-04-22 at 8 43 22 PM

**Comment by cfkoehler** [comment]
> I think it will be necessary to modify default configs for this action.
> 
> The severity defaults to low, I'm not sure we're going to want that? Or maybe we do?
> 
> ```
> with:
>           fail-on-severity: moderate
> ```

I think it is ok to have it set to low. This action will not block a PR if it is from a fork. Either way we can override if it does block. I think we should need to take action to allow even a low severity vulnerability to be merged.

**Review by ldhardy — APPROVED** [review]

**Pull request merged**